### PR TITLE
`Quiz exercises`: Handle partial points in statistics

### PR DESCRIPTION
**Pull request by willergeorg**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I implemented the changes with a good performance and prevented too many database calls.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

The quiz point statistics currently cannot handle partial points of a quiz (i.e. with decimal places). They where not visible at all. This is fixed by this PR. 
Additionally, I fixed a few minor things:

- The average in the quiz statistics (second chart) is now rounded to 2 decimal places in order to prevent too long numbers here.
- I removed the rotation of 45 degree of the datalabels
- The labels are now centered over the corresponding bars

### Description

I adapted the rounding of the results on server side. So far, this was done depending on the course configuration. This is not applicable here, as we distribute the scores in discrete (integer) buckets. Therefore, we now round the score to the nearest natural number in order to make it displayable in the chart.
I additionally adapted the labels on the x axis in order to make the rounding more transparent to the user.

### Steps for Testing

Prerequisites:
- 1 Instructor
- (at least) 1 Student

1. Log in to Artemis as Instructor
2. Create a quiz exercise where it is possible to achieve non-integer points
3. Participate with students in the exercise. Try to achieve partial points.
4. As Instructor, open the quiz statistics. The point statistic is the first chart that is presented to you
5. Compare the individual results displayed to the students with the point distribution of the quiz. The statstic should round the student points and display it respectively in the chart.
6. Check the quiz statistic. Make sure that the average is rounded to one decimal place (if not zero).

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

The problem with the average that occurred (and should be fixed with this PR):

image (4)

Currently, the quiz point statistic is presenting the distribution as if only integers were possible as scores:
image

With this PR, the x axis labels are adapted in order to display the score range summarized in each bar:
image

### CODE REVIEW

**Review by Malyuk-A — DISMISSED** [review]
Code changes look good (I discussed some points with Georg),
Tested with the help of Georg on TS1. Worked as described/expected 👍

File: src/main/java/de/tum/in/www1/artemis/domain/quiz/QuizPointStatistic.java [hunk 1]
```diff
@@ -120,7 +118,7 @@ public void removeOldResult(Double score, Boolean rated) {
      * @param countChange the int-value, which will be added to the Counter and participants
      */
     private void changeStatisticBasedOnResult(double score, Boolean rated, int countChange) {
-        double points = roundScoreSpecifiedByCourseSettings(quiz.getOverallQuizPoints() * (score / 100), quiz.getCourseViaExerciseGroupOrCourseMember());
+        double points = Math.round(quiz.getOverallQuizPoints() * (score / 100));
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/domain/quiz/QuizPointStatistic.java:121]: please add a comment why we do not use `roundScoreSpecifiedByCourseSettings` here

File: src/main/webapp/app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component.ts [hunk 2]
```diff
@@ -126,8 +127,8 @@ export class QuizStatisticComponent extends QuizStatisticsDirective implements O
 
         // add data for the last bar (Average)
         this.backgroundColor.push('#1e3368');
-        this.ratedData.push(this.ratedAverage / this.maxScore);
-        this.unratedData.push(this.unratedAverage / this.maxScore);
+        this.ratedData.push(round(this.ratedAverage / this.maxScore, 2));
```
Comment (krusche) [inline src/main/webapp/app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component.ts:130]: please add a comment, why you don't use the course settings to round here

File: src/main/java/de/tum/in/www1/artemis/domain/quiz/QuizPointStatistic.java (hunk 1 above)
Comment (willergeorg) [reply src/main/java/de/tum/in/www1/artemis/domain/quiz/QuizPointStatistic.java:125]: I added the comments in 3f66b64 in order to make it clear why course dependent rounding is not used at these two spots.

**Review by cremertim — APPROVED** [review]
Tested on ts1, works as expected. Points where rounded properly and the statistics where labeled correctly

**Review by Malyuk-A — APPROVED** [review]
Since my last review, only a couple of comments were added, which seems fine. re-approve

**Review by filip-gregurevic — APPROVED** [review]
Changes look good to me 👍🏼

**Review by Bofan-Zhu — APPROVED** [review]
Tested on ts2. Changes were displayed as expected 👍

**Pull request merged**